hive-metastore/src/test/java/org/apache/iceberg/hive/TestCachedClientPool.java

```diff
@@ -31,11 +38,106 @@ public void testClientPoolCleaner() throws InterruptedException {
     String metastoreUri = hiveConf.get(HiveConf.ConfVars.METASTOREURIS.varname, "");
     CachedClientPool clientPool = new CachedClientPool(hiveConf, Collections.emptyMap());
     HiveClientPool clientPool1 = clientPool.clientPool();
-    Assert.assertTrue(CachedClientPool.clientPoolCache().getIfPresent(metastoreUri) == clientPool1);
+    Supplier<Object> uri = () -> CachedClientPool.URIElement.of(metastoreUri);
+    Assert.assertTrue(
```

Comment:
this should rather check that they are the same instead of using `assertTrue`:

```
Assertions.assertThat(
            CachedClientPool.clientPoolCache()
                .getIfPresent(CachedClientPool.toKey(Collections.singletonList(uri))))
        .isSameAs(clientPool1);
```

Same below